nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2307,11 +2315,49 @@ async fn ip_pool_service_range_remove(
 
 // Disks
 
+#[endpoint {
+    method = GET,
+    path = "/v1/disks",
+    tags = ["disks"],
+}]
+async fn disk_list_v1(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    query_params: Query<params::DiskList>,
+) -> Result<HttpResponseOk<ResultsPage<Disk>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let query = query_params.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let project_lookup =
+            nexus.project_lookup(&opctx, &query.project_selector)?;
```

Comment:
this is invalid when query.pagination is.page matches the WhichPage::Next variant.

Reply:
Fixed in Fixed in [7f608d1](https://github.com/oxidecomputer/omicron/pull/2008/commits/7f608d1d79151efb1ab0c21056f31fd688a92d9c). I also updated the integration tests to verify